sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,362 @@
+import { Account } from "./account";
+import { EncryptedRecord } from "./models/record-provider/encryptedRecord";
+import { OwnedFilter } from "./models/record-scanner/ownedFilter";
+import { OwnedRecord } from "./models/record-provider/ownedRecord";
+import { RecordProvider } from "./record-provider";
+import { RecordPlaintext } from "./wasm";
+import { RecordSearchParams } from "./models/record-provider/recordSearchParams";
+import { RecordsFilter } from "./models/record-scanner/recordsFilter";
+import { RecordsResponseFilter } from "./models/record-scanner/recordsResponseFilter";
+import { RegistrationRequest } from "./models/record-scanner/registrationRequest";
+import { RegistrationResponse } from "./models/record-scanner/registrationResponse";
+import { StatusResponse } from "./models/record-scanner/statusResponse";
+
+type RecordScannerOptions = {
+    url: string;
+    account?: Account;
+    apiKey?: string | { header: string, value: string };
+}
+
+/**
+ * RecordScanner is a RecordProvider implementation that uses the record scanner service to find records.
+ * 
+ * @example
+ * const account = new Account({ privateKey: 'APrivateKey1...' });
+ * 
+ * const recordScanner = new RecordScanner({ url: "https://record-scanner.aleo.org" });
+ * recordScanner.setAccount(account);
+ * recordScanner.setApiKey("your-api-key");
+ * const uuid = await recordScanner.register(0);
+ * 
+ * const filter = {
+ *     uuid,
+ *     filter: {
+ *         program: "credits.aleo",
+ *         records: ["credits"],
+ *     },
+ *     responseFilter: {
+ *         commitment: true,
+ *         owner: true,
+ *         tag: true,
+ *         tag?: boolean;
+ *         sender: true,
+ *         spent: true,
+ *         record_ciphertext: true,
+ *         block_height: true;
+ *         block_timestamp: true;
+ *         output_index: true;
+ *         record_name: true;
+ *         function_name: true;
+ *         program_name: true;
+ *         transition_id: true;
+ *         transaction_id: true;
+ *         transaction_index: true;
+ *         transition_index: true;
+ *     },
+ *     unspent: true,
+ * };
+ * 
+ * const records = await recordScanner.findRecords(filter);
+ */
+class RecordScanner implements RecordProvider {
+    account?: Account;
+    readonly url: string;
+    private apiKey?: { header: string, value: string };
+    private uuid?: string;
+
+    constructor(options: RecordScannerOptions) {
+        this.url = options.url;
+        this.account = options.account;
+        this.apiKey = typeof options.apiKey === "string" ? { header: "X-Provable-API-Key", value: options.apiKey } : options.apiKey;
+    }
+    
```

Comment:
We should add this function for computing the uuid.

```suggestion
    computeUUID(vk: ViewKey): Field {
        // Construct the material needed for the Poseidon oracle.
        const inputs = [Field.newDomainSeparator(RECORD_SCANNER_DOMAIN), vk().toField(), Field.one()]
        // Calculate the uuid.
        const hasher = new Poseidon4();
        return hasher.hash(inputs);
    }
```

You'll need to fold in this [PR 1105](This PR enables domain separators to be built: https://github.com/ProvableHQ/sdk/pull/1105.
) to make this work.

Reply:
done